src/include/duckdb/parser/tableref.hpp

```diff
@@ -37,6 +37,8 @@ class TableRef {
 	shared_ptr<ExternalDependency> external_dependency;
 	//! Aliases for the column names
 	vector<string> column_name_alias;
+	//! Reference recurring table
+	bool is_recurring = false;
```

Comment:
I would prefer not to have this in the TableRef - the TableRef is a central base-class that is used as the basis of all FROM clause elements - and is_recurring is only applicable to recursive CTEs with recursive keys - so this seems like the wrong place for it.

If we switch to using the `recurring.cte` syntax we wouldn't need this flag here I think.